util/types.c

```diff
@@ -78,6 +79,18 @@ char *uint128_t_to_string(nvme_uint128_t val)
 	return str + idx;
 }
 
+char *int128_to_si_string(__u8 *data, __u32 bytes_per_unit)
+{
+	static char str[40];
+	long double val = int128_to_double(data) * bytes_per_unit;
+	const char *suffix = suffix_si_get_ld(&val);
+
+	if (sprintf(str, "(%.2Lf %sB)", val, suffix) > 0)
```

Comment:
Maybe `snprintf` would be a better choice or are you sure that we never write more than 40 chars?